src/book/book.rs

```diff
@@ -215,40 +237,46 @@ fn load_summary_item<P: AsRef<Path>>(
     }
 }
 
-fn load_chapter<P: AsRef<Path>>(
+fn load_chapter<P: AsRef<Path> + Clone>(
     link: &Link,
     src_dir: P,
     parent_names: Vec<String>,
 ) -> Result<Chapter> {
-    debug!("Loading {} ({})", link.name, link.location.display());
-    let src_dir = src_dir.as_ref();
+    let mut ch = if let Some(ref link_location) = link.location {
+        debug!("Loading {} ({})", link.name, link_location.display());
+        let src_dir = src_dir.as_ref();
```

Comment:
I think the `+ Clone` requirements can be removed if this line is moved up so that it has scope in the whole function (and remove the call to `.clone()`).

Reply:
Good catch! :)